internal/git/gitsync_processor.go

```diff
@@ -462,6 +496,48 @@ func NewGitSyncProcessor(ctx context.Context, gitSync *v1alpha1.GitSync, kubeCli
 	return processor, nil
 }
 
+func GetAuthMethod(ctx context.Context, repoUrl string, kubeClient kubernetes.Client, namespace string, repoCred *controllerconfig.GitCredential) (transport.AuthMethod, error) {
+	scheme, err := validations.GetTransportScheme(repoUrl)
```

Comment:
it's weird to call from here into `validations` code. Let's move this method to somewhere in "internal/shared" maybe so both places can get to it